src/components/product/utils/validationSchema.ts

```diff
@@ -241,68 +243,127 @@ export const commonCoreTermsOfSaleValidationSchema = {
-        value ? ethers.utils.isAddress(value) : true
-      ),
-    maxCommits: Yup.string()
-      .required(validationMessage.required)
-      .matches(/^\+?[1-9]\d*$/, "Value must greater than 0"),
-    tokenType: Yup.object({
-      value: Yup.string(),
-      label: Yup.string()
-    })
-      .required(validationMessage.required)
-      .default([{ value: "", label: "" }]),
-    tokenCriteria: Yup.object({
-      value: Yup.string(),
-      label: Yup.string()
-    }).required(validationMessage.required),
-    minBalance: Yup.string().when(["tokenType", "tokenCriteria"], {
-      is: (tokenType: SelectDataProps, tokenCriteria: SelectDataProps) =>
-        tokenType?.value === TOKEN_TYPES[0].value ||
-        tokenType?.value === TOKEN_TYPES[2].value ||
-        (tokenType?.value === TOKEN_TYPES[1].value &&
-          tokenCriteria?.value === TOKEN_CRITERIA[0].value),
-      then: (schema) =>
-        schema
-          .required(validationMessage.required)
-          .matches(
-            /^\+?[1-9]\d*$/,
-            "Min balance must be greater than or equal to 1 (do not include commas/periods)"
-          )
-          .typeError("Value must be an integer greater than or equal to 1")
-    }),
-    tokenId: Yup.string().when(["tokenType", "tokenCriteria"], {
-      is: (tokenType: SelectDataProps, tokenCriteria: SelectDataProps) =>
-        tokenType?.value === TOKEN_TYPES[2].value ||
-        (tokenType?.value === TOKEN_TYPES[1].value &&
-          tokenCriteria?.value === TOKEN_CRITERIA[1].value),
-      then: (schema) =>
-        schema
-          .test(
-            "",
-            "Value must greater than or equal to 0 or a hex value up to 64 chars",
-            (value) => {
-              if (!value) {
-                return false;
+export type TokenGatingValidationSchema = ReturnType<
+  typeof getTokenGatingValidationSchema
+>;
+export const getTokenGatingValidationSchema = ({
+  coreSDK
+}: {
+  coreSDK: CoreSDK;
+}) =>
+  Yup.object({
+    tokenGating: Yup.object({
+      tokenContract: Yup.string()
+        .required(validationMessage.required)
+        .test("FORMAT", "Must be a valid address", (value) =>
+          value ? ethers.utils.isAddress(value) : true
+        )
+        .test("wrongTokenAddress", async function (tokenContract) {
+          const { tokenType } = this.parent;
+          if (
+            tokenType.value &&
+            tokenContract &&
+            ethers.utils.isAddress(tokenContract)
+          ) {
+            try {
+              switch (tokenType.value) {
+                case TokenTypes.erc20:
+                  {
+                    await coreSDK.erc20BalanceOf({
+                      contractAddress: tokenContract,
+                      owner: ethers.constants.AddressZero
+                    });
+                  }
+                  break;
+                case TokenTypes.erc721:
+                  {
+                    await coreSDK.erc721OwnerOf({
```

Comment:
ownerOf() on an ERC1155 (0x6be0e4ccfd378712d19d99065158db1d4f37516f) does not raise the same error code "0x", although the method does not exist
![image](https://github.com/bosonprotocol/interface/assets/7184124/0eb21a8f-4d6b-46a9-bb70-5276a56f48e9).
So setting ERC721 with this address is not raising an error. It should do

Reply:
I guess this is because this contract implement a fallback() method
https://mumbai.polygonscan.com/address/0x0d223d05e1cc4ac20de7fce86bc9bb8efb56f4d4#code
![image](https://github.com/bosonprotocol/interface/assets/7184124/808303e4-855b-4cea-a49e-e2743ceaa1ab)
We should ask @zajck about this

Reply:
see https://github.com/bosonprotocol/interface/pull/908#discussion_r1387773633